### PR TITLE
chore: fix Windows compilation process

### PR DESCRIPTION
**Pull request by Spycsh**

### CODE REVIEW

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -432,7 +433,8 @@ default_ciphers() -> [
     ] ++ psk_ciphers().
 
 psk_ciphers() -> [
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:435]: add a os:type check here to only exclude it when it's `win32`

File: bin/emqx.cmd [hunk 2]
```diff
@@ -15,21 +15,33 @@
 
 :: Set variables that describe the release
 @set rel_name=emqx
-@set rel_vsn={{ release_version }}
+@set rel_vsn=5.0-alpha.5-5b75fdd1
```
Comment (zmstone) [inline bin/emqx.cmd:18]: `{{ release_version }}` is rendered by release script.

File: bin/emqx.cmd [hunk 3]
```diff
@@ -15,21 +15,33 @@
...
 @set REL_VSN=%rel_vsn%
-@set erts_vsn={{ erts_vsn }}
-@set erl_opts={{ erl_opts }}
+@set erts_vsn=11.1
```
Comment (zmstone) [inline bin/emqx.cmd:20]: this is also rendered by release script.

File: apps/emqx_machine/src/emqx_machine_schema.erl [hunk 4]
```diff
@@ -284,6 +284,15 @@ fields("node") ->
        sc(ref("cluster_call"),
           #{}
          )}
+    , {"etc_dir",
+       sc(string(),
+          #{converter => fun(EtcDir) -> case EtcDir of
+                                          "_" -> filename:join([code:lib_dir(), "..", "etc"]);
```
Comment (zmstone) [inline apps/emqx_machine/src/emqx_machine_schema.erl:290]: why `"_"` ?

File: rebar.config.erl [hunk 5]
```diff
@@ -215,7 +215,7 @@ overlay_vars_rel(RelType) ->
 overlay_vars_pkg(bin) ->
     [ {platform_bin_dir, "bin"}
     , {platform_data_dir, "data"}
-    , {platform_etc_dir, "etc"}
+    , {platform_etc_dir, "_"}
```
Comment (zmstone) [inline rebar.config.erl:218]: should not change this.

File: apps/emqx_machine/src/emqx_machine_schema.erl [hunk 6]
```diff
@@ -285,7 +285,7 @@ fields("node") ->
           #{}
          )}
     , {"etc_dir",
-       sc(string(),
+       sc(list_to_binary(string()),
```
Comment (zmstone) [inline apps/emqx_machine/src/emqx_machine_schema.erl:288]: you can't do list_to_binary here as schema.

Comment (zmstone) [reply apps/emqx_machine/src/emqx_machine_schema.erl:288]: the `string()` function is imported from `typerefl`
it actually means `typerefl:string()` which returns a tuple, but not a list.

File: apps/emqx/src/emqx_schema.erl [hunk 7]
```diff
@@ -967,22 +967,29 @@ tls_vsn(<<"tlsv1.2">>) -> 'tlsv1.2';
...
+        "DHE-DSS-AES256-SHA256", "AES256-GCM-SHA384", "AES256-SHA256",
+        "ECDHE-ECDSA-AES128-GCM-SHA256", "ECDHE-RSA-AES128-GCM-SHA256",
+        "ECDHE-ECDSA-AES128-SHA256", "ECDHE-RSA-AES128-SHA256", "ECDH-ECDSA-AES128-GCM-SHA256",
+        "ECDH-RSA-AES128-GCM-SHA256", "ECDH-ECDSA-AES128-SHA256", "ECDH-RSA-AES128-SHA256",
+        "DHE-DSS-AES128-GCM-SHA256", "DHE-DSS-AES128-SHA256", "AES128-GCM-SHA256", "AES128-SHA256",
+        "ECDHE-ECDSA-AES256-SHA", "ECDHE-RSA-AES256-SHA", "DHE-DSS-AES256-SHA",
+        "ECDH-ECDSA-AES256-SHA", "ECDH-RSA-AES256-SHA", "AES256-SHA", "ECDHE-ECDSA-AES128-SHA",
+        "ECDHE-RSA-AES128-SHA", "DHE-DSS-AES128-SHA", "ECDH-ECDSA-AES128-SHA",
+        "ECDH-RSA-AES128-SHA", "AES128-SHA"
+        ],
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:988]:
```suggestion
        ] ++ psk_ciphers(os:type()).

psk_ciphers({win32, _}) -> [];
psk_ciphers(_) -> [        
```

then we do not have to change that many lines of code.

File: apps/emqx_authz/src/emqx_authz_schema.erl [hunk 8]
```diff
@@ -177,3 +173,16 @@ connector_fields(DB, Fields) ->
...
+        true -> MaybeRelativePath;
+        false -> emqx_system:etc_file(MaybeRelativePath)
+    end.
+
+validate_etc_file(MaybeRelativePath) ->
+    AbsPath = etc_file(MaybeRelativePath),
+    case filelib:is_file(AbsPath) of
+        true -> ok;
+        _ -> {error, "File does not exist"}
+    end.
```
Comment (zmstone) [inline apps/emqx_authz/src/emqx_authz_schema.erl:188]: please configure your IDE to ensure a line-break at the last line.

File: apps/emqx_machine/src/emqx_system.erl [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_system).
+-export([etc_file/1]).
+
+etc_file(RelativePathToFile) ->
+    filename:join([emqx_config:get([node, etc_dir]), RelativePathToFile]).
```
Comment (zmstone) [inline apps/emqx_machine/src/emqx_system.erl:21]: ensure last line-break

File: apps/emqx_machine/src/emqx_machine_schema.erl [hunk 10]
```diff
@@ -285,8 +285,10 @@ fields("node") ->
           #{}
          )}
     , {"etc_dir",
-       sc(typerefl:string(),
-          #{converter => fun(EtcDir) -> case list_to_binary(EtcDir) of
+       sc(string(),
+          #{converter => fun(EtcDir) -> 
+             io:format("yyyyyyyy:~p~n",[EtcDir]),
+             case list_to_binary(EtcDir) of
                                           <<"_">> ->
```
Comment (zmstone) [inline apps/emqx_machine/src/emqx_machine_schema.erl:292]:
```suggestion
                                          <<"">> -> 
```

File: apps/emqx/src/emqx_schema.erl [hunk 11]
```diff
@@ -972,22 +972,29 @@ tls_vsn(<<"tlsv1.2">>) -> 'tlsv1.2';
...
-    "ECDHE-ECDSA-AES128-SHA256", "ECDHE-RSA-AES128-SHA256", "ECDH-ECDSA-AES128-GCM-SHA256",
-    "ECDH-RSA-AES128-GCM-SHA256", "ECDH-ECDSA-AES128-SHA256", "ECDH-RSA-AES128-SHA256",
-    "DHE-DSS-AES128-GCM-SHA256", "DHE-DSS-AES128-SHA256", "AES128-GCM-SHA256", "AES128-SHA256",
-    "ECDHE-ECDSA-AES256-SHA", "ECDHE-RSA-AES256-SHA", "DHE-DSS-AES256-SHA",
-    "ECDH-ECDSA-AES256-SHA", "ECDH-RSA-AES256-SHA", "AES256-SHA", "ECDHE-ECDSA-AES128-SHA",
-    "ECDHE-RSA-AES128-SHA", "DHE-DSS-AES128-SHA", "ECDH-ECDSA-AES128-SHA",
-    "ECDH-RSA-AES128-SHA", "AES128-SHA"
-    ] ++ psk_ciphers().
+default_ciphers() -> 
+    NormalCiphers =
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:976]: there should be no need for this Variable,
and the indentation changes in the following lines can be reverted.

**Comment by Spycsh** [comment]
@zmstone commits are somewhat messed up here because I do not really know that this pull request can track my commit on my local github repo until now... it seems hard to me to find the one that really correctly change the etc_dir and separate it from the windows bat file update that also enclosed in this PR. One stupid method is that maybe I can refork this repository and make two pure new PRs about the two separate things. WDYT

File: apps/emqx/src/emqx_schema.erl [hunk 12]
```diff
@@ -1025,25 +1025,28 @@ tls_vsn(<<"tlsv1.1">>) -> 'tlsv1.1';
...
-    "ECDHE-ECDSA-AES256-SHA", "ECDHE-RSA-AES256-SHA", "DHE-DSS-AES256-SHA",
-    "ECDH-ECDSA-AES256-SHA", "ECDH-RSA-AES256-SHA", "AES256-SHA", "ECDHE-ECDSA-AES128-SHA",
-    "ECDHE-RSA-AES128-SHA", "DHE-DSS-AES128-SHA", "ECDH-ECDSA-AES128-SHA",
-    "ECDH-RSA-AES128-SHA", "AES128-SHA"
-    ] ++ psk_ciphers().
-
-psk_ciphers() -> [
-        "PSK-AES128-CBC-SHA", "PSK-AES256-CBC-SHA", "PSK-3DES-EDE-CBC-SHA", "PSK-RC4-SHA"
-    ].
+        "TLS_AES_256_GCM_SHA384", "TLS_AES_128_GCM_SHA256", "TLS_CHACHA20_POLY1305_SHA256",
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:1028]: please revert indentation changes

File: apps/emqx_machine/src/emqx_machine_schema.erl [hunk 13]
```diff
@@ -310,27 +310,27 @@ fields("node") ->
...
-                case filelib:is_dir(Path) of
-                   true -> 
-                      ok;
-                   false -> 
-                      error({not_dir, Path})
-                end
-               end
-            }
-         )}
+   %  , {"etc_dir",
```
Comment (zmstone) [inline apps/emqx_machine/src/emqx_machine_schema.erl:313]: we may still need this config, please add it back (without converter and validator).

**Comment by zmstone** [comment]
closed in favour of https://github.com/emqx/emqx/pull/5822

**Pull request closed without merging**